src/Model/Contact.php

```diff
@@ -2737,17 +2743,36 @@ public static function addRelationship(array $importer, array $contact, array $d
 
 				self::update($fields, $condition);
 
+				DI::notify()->createFromArray([
+					'type'  => Notification\Type::INTRO,
+					'otype' => Notification\ObjectType::INTRO,
+					'verb'  => ($sharing ? Activity::FRIEND : Activity::FOLLOW),
+					'uid'   => $user['uid'],
+					'cid'   => $contact_record['id'],
+					'link'  => DI::baseUrl() . '/contact/' . $contact['id'],
+				]);
+
 				return true;
 			}
 		}
 
 		return null;
 	}
 
+	/**
+	 * Update the local relationship when a local user loses a follower
+	 *
+	 * @param array $contact User-specific contact (uid != 0) array
+	 * @throws HTTPException\InternalServerErrorException
+	 * @throws \ImagickException
+	 */
 	public static function removeFollower(array $contact)
 	{
+		// Remove follow request notifications
+		DI::notify()->deleteForUserByContactId($contact['uid'], $contact['id'], [Type::INTRO]);
```

Comment:
Please ensure to apply the changes also to the `notification`  table. We really should drop this old `notify` table and finally do everything with the `notification` table.